main.c

```diff
@@ -182,36 +190,66 @@ static const struct wl_pointer_listener pointer_listener = {
 	.axis = noop,
 };
 
+static void keyboard_handle_keymap(void *data, struct wl_keyboard *wl_keyboard,
+		uint32_t format, int32_t fd, uint32_t size) {
+	struct slurp_seat *seat = data;
+	switch (format) {
+	case WL_KEYBOARD_KEYMAP_FORMAT_NO_KEYMAP:
+		seat->xkb_keymap = xkb_keymap_new_from_names(seat->state->xkb_context, NULL, XKB_KEYMAP_COMPILE_NO_FLAGS);
+		break;
+	case WL_KEYBOARD_KEYMAP_FORMAT_XKB_V1:;
+		void *buffer;
+		if ((buffer = mmap(NULL, size - 1, PROT_READ, MAP_SHARED, fd, 0)) == MAP_FAILED) {
```

Comment:
Prefer `MAP_PRIVATE` to `MAP_SHARED`.

Reply:
according to http://man7.org/linux/man-pages/man2/mmap.2.html, `MAP_PRIVATE` uses copy-on-write which (I suppose) has some overhead. Also we don't really need this feature since we're not 1. writing to that memory and 2. accessing that memory from other processes

Reply:
See https://gitlab.freedesktop.org/wayland/wayland/-/blob/master/protocol/wayland.xml#L2182 for the reason why we should use `MAP_PRIVATE`